ocaml/typing/typedecl.ml

```diff
@@ -426,7 +426,13 @@ let transl_labels ~new_var_jkind ~allow_unboxed env univars closed lbls kloc =
           | Immutable -> Immutable
           | Mutable -> Mutable Mode.Alloc.Comonadic.Const.legacy
          in
-         let modalities = Typemode.transl_modalities mut modalities in
+         let mut_mod =
+          if Types.is_mutable mut then
+            not (Builtin_attributes.has_no_mutable_implied_modalities attrs)
+          else
+            false
```

Comment:
This is intentional - I want to emphesizes the fact that the attribute will not be looked at all if the field is not mutable.

The document of `&&` specifies the ordering, but I still tend to avoid relying on that.